cypress/components/flat-table/flat-table.test.js

```diff
@@ -4658,4 +4658,275 @@ context("Tests for Flat Table component", () => {
+      CypressMountWithProviders(
+        <FlatTableCellColSpanComponent width="500px" />
+      );
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table with rowSpan set to make cells span 3 rows for accessibility tests", () => {
+      CypressMountWithProviders(
+        <FlatTableCellRowSpanComponent width="500px" />
+      );
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table with FlatTableFooterComponent for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableFooterComponent />);
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table with FlatTableCustomPaddingComponent for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableCustomPaddingComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table Header and Cells with FlatTableVerticalBordersComponent for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableVerticalBordersComponent />);
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table with custom horizontal borders for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableHorizontalBordersComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    // a11y error!color-contrast on 1 Node. Ticket FE-5766 logged to investigate
+    it.skip("should render Flat Table with custom color row for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableColorRowSelectableComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    // a11y error!color-contrast on 1 Node. Ticket FE-5766 logged to investigate
+    it.skip("should render Flat Table rows with the mouse for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableHighlightableComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table Row Header with truncated string in header for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableTruncateHeaderComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table Row Header with title for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableTitleAlignComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should sort Flat Table %s column in %s order with mouse click for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableSortingComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table with expandable rows expanded by mouse, subrows not accessible for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableNoAccSubRowComponent />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should render Flat Table expandable by first column only by mouse for accessibility tests", () => {
+      CypressMountWithProviders(<FlatTableFirstColExpandableComponent />);
+
+      cy.checkAccessibility();
+    });
```

Comment:
Given all of the expandable rows are closed when we perform the checks, it might make sense to expand one of the rows first and then do the checks, as the rendered html markup will be different. 

```suggestion
    it("when first column of expandable row is opened, rendered Flat Table passes accessibility checks", () => { 
      CypressMountWithProviders(<FlatTableFirstColExpandableComponent />);

      flatTableCell(0).focus().trigger("keydown", keyCode("Space"));

      cy.checkAccessibility();
    });
```